eudemo/eudemo/equatorial_port.py

```diff
@@ -0,0 +1,200 @@
+from bluemira.base.components import Component, PhysicalComponent
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.tools import (
+    boolean_fuse,
+    circular_pattern,
+    extrude_shape,
+    make_polygon,
+)
+from bluemira.geometry.wire import BluemiraWire
+
+
+class EquatorialPort(ComponentManager):
+    """
+    Wrapper around a Equatorial Port component tree
+    """
+
+    def xz_boundary(self) -> BluemiraWire:
+        """Returns a wire defining the x-z boundary of the Equatorial Port"""
+        return (
+            self.component.get_component("xz")
+            .get_component(EquatorialPortBuilder.KOZ)
+            .shape.boundary[0]
+        )
+
+
+@dataclass
+class EquatorialPortDesignerParams(ParameterFrame):
+    """
+    Equatorial Port Designer parameters
+    """
+
+    x_ib: Parameter[float]
+    x_ob: Parameter[float]
+    ep_height: Parameter[float]
+
+
+@dataclass
+class EquatorialPortBuilderParams(ParameterFrame):
+    """
+    Equatorial Port builder parameters
+    """
+
+    n_EP: Parameter[int]
+    ep_coordinates: Parameter[ndarray]
+    ep_r_corner: Parameter[float]
+
+
+class EquatorialPortDesigner(Designer):
+    """
+    Equatorial Port Designer
+    """
+
+    param_cls: Type[EquatorialPortDesignerParams] = EquatorialPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortDesignerParams],
+        build_config: Union[Dict, None],
+    ):
+        super().__init__(params, build_config)
+
+    def run(self) -> BluemiraWire:
+        """
+        Design the xz keep-out zone profile of the equatorial port
+        """
+
+        x_i = self.params.x_ib.value
+        x_o = self.params.x_ob.value
+        z_h = self.params.ep_height.value
+
+        x = (x_i, x_o, x_o, x_i)
+        z = (-z_h / 2, -z_h / 2, z_h / 2, z_h / 2)
+
+        ep_boundary = BluemiraFace(
+            make_polygon({"x": x, "y": 0, "z": z}, label="koz", closed=True)
+        )
+        return ep_boundary
+
+
+class EquatorialPortBuilder(Builder):
+    """
+    Equatorial Port Builder
+    """
+
+    NAME = "Equatorial Port"
+    KOZ = "koz"
+    param_cls: Type[EquatorialPortBuilderParams] = EquatorialPortBuilderParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortBuilderParams],
+        build_config: Union[Dict, None],
+        xz_profile: BluemiraFace,
+    ):
+        super().__init__(params, build_config)
+        self.xz_profile = xz_profile
+
+    def build(self) -> Component:
+        """
+        Build the equatorial port component.
+        """
+
+        # TODO: Integrate corner radii
+        self.r_rad = self.params.ep_r_corner.value
+        self.coords = self.params.ep_coordinates.value

```

Comment:
having to specify all the coords for the user is a big ask. I think this needs to be the inboard x coordinate, the outboard x coordinate and the z offset of a single castellation. We should be able to build up the castellation from that.

Reply:
I agree, but might keep the castellation method handy for future reference.

As well as the specified coordinates, wouldn't it also be necessary to specify the x-position of the castellation? Or is that inferred as halfway or at the outer boundary of the VV?

Reply:
Yes true, so 2 x coordinates one for inside and one for outside of a given vessel. 2 offsets for a single castellation 1 in z and 1 in x. 
I would have them as separate params